src/pages/Profile.js

```diff
@@ -40,31 +40,92 @@ function Profile(props) {
         getUserInfo()
     }, [])
 
+    function settingEditToggle(){
+        setEditToggle(!editToggle)
+    }
+
+    //--This function is essentially the main "PUT" request for editing the Account info.
+    async function editAccountInfoRequest(accountPayload) {
+        let response = {};
+
+            response = await apiUtil("put", `/users/${userState.user.id}`, {dispatch: dispatch, navigate: navigate}, accountPayload)
+            console.log(response);
+            setError(response.error)
+            setMessage(response.message)
+        
+    }
+
+    //this function calls the api call, as well as sets the states all back to 
+    //being blank.
+    function sendEditedAccountInfo(){
+
+        const newAccountInfo = {
+            email: email,
+            firstName: firstName,
+            lastName: lastName
+        }
+
+        if (currentPassword != "")
+            newAccountInfo.oldPassword = currentPassword
+        if (newPassword != "")
+            newAccountInfo.rawPassword = newPassword
+        if (confirmedPassword != "")
+            newAccountInfo.confirmedPassword = confirmedPassword
+
+        editAccountInfoRequest(newAccountInfo);
+
+        setCurrentPassword("")
```

Comment:
it looks like this one might also need to be wrapped in an if statement or something